src/test-jdk17/java/com/fasterxml/jackson/databind/records/RecordWithJsonUnwrappedTest.java

```diff
@@ -0,0 +1,32 @@
+package com.fasterxml.jackson.databind.records;
+
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.struct.TestUnwrappedWithJsonCreator;
+
+public class RecordWithJsonUnwrappedTest extends BaseMapTest {
+    record RecordWithJsonUnwrapped(String unrelated, @JsonUnwrapped Inner inner) {
+    }
+
+    record Inner(String property1, String property2) {
+    }
+
+    private final ObjectMapper MAPPER = newJsonMapper();
+
+    /*
+    /**********************************************************************
+    /* Tests, JsonUnwrapped deserialization
+    /**********************************************************************
+     */
+
+    public void testUnwrappedWithRecord() throws Exception
+    {
+        String json = "{\"unrelated\": \"unrelatedValue\", \"property1\": \"value1\", \"property2\": \"value2\"}";
+        RecordWithJsonUnwrapped outer = MAPPER.readValue(json, RecordWithJsonUnwrapped.class);
```

Comment:
Works, but might make sense to first serialize from instance, then read back (round-trip). To make both directions work (I realize that this PR is just for deserialization but still)

Reply:
Changed it to a round-trip. Should I also change the tests in `UnwrappedWithCreatorTest`? Since there are already similar tests for serializing regular classes.